File: JAVApb5/ConcursModel/src/main/java/concurs/domain/Participant.java
package concurs.domain;

import java.io.Serializable;
import java.util.Objects;

public class Participant implements Entity<Long>, Serializable {
    private long id;
    private String nume;
    private String prenume;
    private int varsta;
    public Participant() {

    }
    public Participant(String nume, String prenume, int varsta) {
        this.nume = nume;
        this.prenume = prenume;
        this.varsta = varsta;
    }

    public String getNume() {
        return nume;
    }

    public void setNume(String nume) {
        this.nume = nume;
    }

    public String getPrenume() {
        return prenume;
    }

    public void setPrenume(String prenume) {
        this.prenume = prenume;
    }

    public int getVarsta() {
        return varsta;
    }

    public void setVarsta(int varsta) {
        this.varsta = varsta;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Participant that = (Participant) o;
        return varsta == that.varsta &&
                Objects.equals(nume, that.nume) &&
                Objects.equals(prenume, that.prenume);
    }

    @Override
    public int hashCode() {
        return Objects.hash(nume, prenume, varsta);
    }

    @Override
    public String toString() {
        return "Participant{" +
                "id"+id+
                "nume='" + nume + '\'' +
                ", prenume='" + prenume + '\'' +
                ", varsta=" + varsta +
                '}';
    }

    @Override
    public Long getId() {
       return id;
    }

    @Override
    public void setId(Long id) {
    this.id=id;
    }
}

File: CSharp/domain/Participant.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CSharp.domain
{
    public class Participant:Entity<long>
    {   public  long Id { get; set; }
        public String Nume { get; set; }
        public String Prenume { get; set; }
        public int Varsta { get; set; }
        
    
        public Participant(string nume, string prenume, int varsta)
        {
            Nume = nume;
            Prenume = prenume;
            Varsta = varsta;
        }
     public Participant(){}
        public override bool Equals(object obj)
        {
            return obj is Participant participant &&
                   Nume == participant.Nume &&
                   Prenume == participant.Prenume &&
                   Varsta == participant.Varsta;
        }

        /*public override int GetHashCode()
        {
            return base.GetHashCode();
        }
*/
        public override string ToString()
        {
            return string.Format("[Participant: Id={0}, Nume={1},Prenume={2},Varsta={3}]", Id, Nume,Prenume,Varsta);
        }
       
    }
}
